api/app/controllers/spree/api/v2/base_controller.rb

```diff
@@ -0,0 +1,48 @@
+module Spree
+  module Api
+    module V2
+      class BaseController < ActionController::API
+        include CanCan::ControllerAdditions
+
+        private
+
+        def spree_current_store
+          @spree_current_store ||= Spree::Store.current(request.env['SERVER_NAME'])
+        end
+
+        def spree_current_user
+          @spree_current_user ||= Spree.user_class.find_by(id: doorkeeper_token.resource_owner_id) if doorkeeper_token
+        end
+
+        def spree_authorize!(action, subject, *args)
+          authorize!(action, subject, *args)
+        end
+
+        # Needs to be overriden so that we use Spree's Ability rather than anyone else's.
+        def current_ability
+          @current_ability ||= Spree::Ability.new(spree_current_user)
+        end
+
+        def order_token
+          request.headers['X-Spree-Order-Token'] || params[:order_token]
+        end
+
+        def current_order
+          @current_order || find_or_create_current_order
+        end
+
+        def find_or_create_current_order
+          @current_order = Spree::CurrentOrderFinder.new(user: spree_current_user,
+                                                         store: spree_current_store,
+                                                         guest_token: order_token,
+                                                         currency: Spree::Config[:currency]).execute
```

Comment:
you can use `current_currency` method here

Reply:
I can't since I didn't include old controller helpers. We want to break dependency on older versions so we have to reimplement it somehow here(without using old helpers)

Reply:
I've added current_currency method so anyone can overwrite it in their BaseController from now.